### ISSUE TITLE
Customer Sales Tax Exemptions not work

### ISSUE THREAD

**Issue by APanteleychuk** [taxjar/taxjar-magento2-extension#155]
### Description
When I set the TaxJar Exemption Type field (wholesale) for a specific customer, I still have an additional tax on the checkout.

### Steps to Reproduce

1. In admin panel need set in specific customer TaxJar Exemption field - wholesale;
    (Like here https://www.taxjar.com/guides/integrations/magento2/#sales-tax-calculations)
2. Need to login for this customer
3. Add some (simple, bundle, configurable) product in the cart
4. Go in checkout page.

### Expected Result
Exempt from sales tax calculations at checkout

### Actual Result
Not exempt from sales tax calculations at checkout

### Versions
- Magento 2.2.4(ЕЕ)
- PHP 7.1.33
- Taxjar 1.4.2

**Comment by ryanreeves-taxjar** [comment]
Hi @APanteleychuk!  Can you please update to the latest version of our extension (v1.5.6) and confirm this problem still exists?

**Comment by ryanreeves-taxjar** [comment]
Closing this issue as inactive, but if the problem persists please feel free to reach out again!

**Comment by perryholden** [comment]
This does happen with version 1.6.0. What is happening is when a customer is set to tax exempt, then TaxJar returns the native Magento tax collector results. If a user has fallback tax rules set up in Magento, then those rules are triggered.

Basically what happens is that if `\Taxjar\SalesTax\Model\Smartcalcs::_isCustomerExempt` returns true, then the response object is never created, so it returns status 204 (see `\Taxjar\SalesTax\Model\Smartcalcs::getResponse` for how this is done). Afterwards, in `\Taxjar\SalesTax\Model\Tax\Sales\Total\Quote\Tax::collect` line 145, where it checks for a valid response, if it's not 200, then it returns the native Magento tax collector results (see line 178).

Basically, this extension needs to be updated that if a customer is tax exempt, it also skips native tax calculations too. Please re-open.

**Comment by fastdivision** [comment]
Thanks for sharing this behavior @perryholden. You're right, if there are native tax rules for a customer tax class with exempt customers then tax may still be applied. On the other hand, our backup rate feature allows you to select specific customer tax classes as shown below. If you wanted to skip native calculations for exempt customers, you would remove the tax rule assigned to the exempt customer tax class. Our fallback to native calculations provides a lot of flexibility for merchants to skip TaxJar API calculations and handle custom rates or taxability for unique scenarios, so this kind of change would have to be carefully considered first.

![magento-backup-rates-config](https://user-images.githubusercontent.com/1137184/110556139-c5913180-80f2-11eb-82a5-49e3a0b80254.png)
